File: clinic-management-system/src/main/java/com/cms/clinic/entity/Slot.java
package com.cms.clinic.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalTime;
import java.util.Date;

@Entity
@Table(name = "slot", schema = "dbo")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Slot {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long slotId;
    private LocalTime time;
    private Long patientId;
    private Long doctorId;
    @JsonFormat(pattern = "yyyy-MM-dd", shape = JsonFormat.Shape.STRING)
    private Date dateTime = new Date();
//    @OneToOne
//    @JoinColumn(name = "appointment_id")
//    private Appointment appointment;
}

File: clinic-angular/src/app/models/slot.ts
import { LocalTime } from './localTime';

export class Slot {
  // slotId?: number;
  // time?: Date;
  // dateTime?: Date;
  // patientId?: number;
  // doctorId?: number;

  slotId?: number;
  time?: LocalTime;
  patientId?: number;
  doctorId?: number;
  dateTime?: string;
}
